Astral Pathfinder/Game/UIManager.cpp

```diff
@@ -73,12 +69,104 @@ void UIManager::render(Game::State *gameState) {
+  if(gameState->planetCollided) currentPlanetInfo.render(pm->getCurrentPlanet().getPopulation());
 }
 
 // MARK: - UIManager Methods
 
 void UIManager::setSelectedPlanet(Planet p) {
-  planetInfo.setUiTextures(p);
+  selectedPlanetInfo.setUiTextures(p);
+}
+
+void UIManager::setCurrentPlanet(Planet p) {
+  currentPlanetInfo.setUiTextures(p);
+}
+
+void UIManager::handleMouseDown(Game::State *gs, PlanetManager *pm) {
+  // If mouse button not pressed down, don't check for slider movement
+  if(!gs->mouseDown) return;
+  
+  if(gs->mouseDown)
+    window = checkClickedArea(gs->clickLocation);
+  
+  // Current Planet Window
+  if(window == 1) {
+    // If down, but not dragging, check if slider was clicked
+    if(gs->mouseDown && gs->activeSlider == gs->State::inactive) {
+      if(currentPlanetInfo.checkClick(gs->clickLocation) == 1)
+        gs->activeSlider = gs->State::currentOne;
+      
+      if(currentPlanetInfo.checkClick(gs->clickLocation) == 2)
+        gs->activeSlider = gs->State::currentTwo;
+    }
+  
+    // If so, check mouse movement and adjust slider appropriately
+    if(gs->activeSlider == gs->State::currentOne) {
+      bool movement = currentPlanetInfo.moveSlider(gs);
+      int percent = currentPlanetInfo.getSliderPercent();
+      
+      if(movement) {
+        pm->setPlanetDepoPercent(100-percent, window);
+        pm->setPlanetFertPercent(percent, window);
+      }
+    }
+  
+    if(gs->activeSlider == gs->State::currentTwo) {
+      bool movement = currentPlanetInfo.moveSlider(gs);
+      int percent = currentPlanetInfo.getSliderPercent();
+      
+      if(movement) {
+        pm->setPlanetInfraPercent(100-percent, window);
+        pm->setPlanetReservePercent(percent, window);
+      }
+    }
+  }
+  
+  // Selected Planet Window
+  if(window == 2) {
+    // If down, but not dragging, check if slider was clicked
+    if(gs->mouseDown && gs->activeSlider == gs->State::inactive) {
+      if(selectedPlanetInfo.checkClick(gs->clickLocation) == 1)
+        gs->activeSlider = gs->State::selectOne;
+      
+      if(selectedPlanetInfo.checkClick(gs->clickLocation) == 2)
+        gs->activeSlider = gs->State::selectTwo;
+    }
+    
+    // If so, check mouse movement and adjust slider appropriately
+    if(gs->activeSlider == gs->State::selectOne) {
+      bool movement = selectedPlanetInfo.moveSlider(gs);
+      int percent = selectedPlanetInfo.getSliderPercent();
+      
+      if(movement) {
+        pm->setPlanetDepoPercent(100-percent, window);
+        pm->setPlanetFertPercent(percent, window);
+      }
+    }
+    
+    if(gs->activeSlider == gs->State::selectTwo) {
+      bool movement = selectedPlanetInfo.moveSlider(gs);
+      int percent = selectedPlanetInfo.getSliderPercent();
+      
+      if(movement) {
+        pm->setPlanetInfraPercent(100-percent, window);
+        pm->setPlanetReservePercent(percent, window);
+      }
+    }
+  }
 }
 
+int UIManager::checkClickedArea(SDL_Point p) {
```

Comment:
Maybe use enum to clarify what is being returned?

Reply:
Done